modules/blockbuilder/blockbuilder.go

```diff
@@ -233,39 +233,49 @@ func (b *BlockBuilder) consume(ctx context.Context) (time.Duration, error) {
 		if !p.hasRecords() { // No records, we can skip the partition
 			continue
 		}
-		lastRecordTs, lastRecordOffset, err := b.consumePartition(ctx, p)
+		lastRecordTs, commitOffset, err := b.consumePartition(ctx, p)
 		if err != nil {
 			return 0, err
 		}
 		ps[i].lastRecordTs = lastRecordTs
-		ps[i].startOffset = lastRecordOffset
+		ps[i].commitOffset = commitOffset
 	}
 
+	sort.Slice(ps, func(i, j int) bool {
+		now := time.Now()
+		return now.Sub(ps[i].lastRecordTs) > now.Sub(ps[j].lastRecordTs)
```

Comment:
This is comparing 2 deltas, which I think can be simplified to comparing the timestamps directly `ps[i].lastRecordTs.Before(ps[j].lastRecordTs)`